dbdimp.c

```diff
@@ -1827,7 +1827,7 @@ int dbd_discon_all (SV *drh, imp_drh_t *imp_drh) {
   dTHX;
```

Comment:
Unfortunately I don't think this subroutine is actually called any more.  It was removed in 2003 as per commit history?  I think that function needs to be re-visited and must be called to call the `mysql_library_end()` sub.  This is actually quite a bad situation as `DBD::mysql` is currently not cleaning up at the right point, at least in my reading of the situation.  I am fairly new to looking at this, so I defer to those with more insight, however empirically I know that calling `mysql_library_end` in every call to DESTROY is causing issues when the `openssl_plugin` is unloaded.

Reply:
I've added this in `dbdimp.h` and it seems to get called. So this is one step forward.

Looks like 3921e6bee0e5480c9490cdff66c4c190c085bb44 was the commit that removed this.

Reply:
Just curious how you verified that it is being called?  I did the same thing by adding it to `dbdimp.h` and then adding some print statements (I know that is pretty crude) but did not see any output.  Adding print statements at other points in the C code produces messages on my console, so I'm pretty sure that if it had been called I would have seen the messages.  OTOH - I may have made some other error that prevented me from seeing the message.  In any case, I do think this  is where `mysql_library_end` needs to be called but would like to know how we can verify that.  Thanks so much...nice work.

Reply:
Adding something like `croak("this is called")`  seems to work for me and so does GDB with adding a breakpoint ( `b mysql_discon_all` iirc).

Reply:
perfect - thank you